crates/nu-cli/src/completions/completer.rs

```diff
@@ -829,6 +861,12 @@ impl NuCompleter {
             _ => (),
         }
 
+        // try argument dynamic completion defined by Command first.
+        let value_completion_result = self.process_completion(value_completion, ctx);
+        if !value_completion_result.is_empty() {
+            return value_completion_result;
+        }
+
```

Comment:
Since I'll eventually merge `argument_completion_helper` into a new `ArgValueCompleter` (both dynamic and static). I think it makes more sense to move this before the calling of `argument_completion_helper`.

Also we need to pay attention to whether we want to, or how to enable fallback ability to this as well as the flag value dynamic completion.

Reply:
Sure, I moved the logic outside `argument_completion_helper`.